evm/bridge-token-factory/contracts/BridgeTokenFactoryWormhole.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.24;
+
+import {BridgeTokenFactory} from "./BridgeTokenFactory.sol";
+import "./Borsh.sol";
+
+interface IWormhole {
+    function publishMessage(
+        uint32 nonce,
+        bytes memory payload,
+        uint8 consistencyLevel
+    ) external payable returns (uint64 sequence);
+}
+
+contract BridgeTokenFactoryWormhole is BridgeTokenFactory {
+    IWormhole private _wormhole;
+    uint8 private _consistencyLevel;
+    uint32 public wormholeNonce;
+
+    function initializeWormhole(
+        address _tokenImplementationAddress,
+        address _nearBridgeDerivedAddress,
+        address wormholeAddress,
+        uint8 consistencyLevel
+    ) external initializer {
+        initialize(_tokenImplementationAddress, _nearBridgeDerivedAddress);
+        _wormhole = IWormhole(wormholeAddress);
+        _consistencyLevel = consistencyLevel;
+    }
+
+    function depositExtension(BridgeDeposit memory bridgeDeposit) internal override {
+        _wormhole.publishMessage(
```

Comment:
this method is payable 
https://github.com/wormhole-foundation/wormhole/blob/a3284d66d5da7c0480262d5f7c93d7cc8322f6a1/ethereum/contracts/bridge/Bridge.sol#L358

Reply:
f781851ad9ce65355f64c75da1ef143f5338411d